libs/partners/mistralai/langchain_mistralai/chat_models.py

```diff
@@ -216,8 +247,15 @@ def _convert_message_to_mistral_chat_message(
                 if _id := tc.get("id"):
                     chunk["id"] = _id
                 tool_calls.append(chunk)
+        elif message.tool_calls or message.invalid_tool_calls:
```

Comment:
Should this be prioritized over additional_kwargs? The latter should be the source of truth if swapping providers etc

Reply:
ah yeah if we have a provider that (1) formats tool_calls into additional_kwargs, and (2) does not use OpenAI-style formatting then this will break. can update.

Reply:
@efriis see above - is it OK to lose the behavior maintained in unit tests, in which applying `_convert_message_to_mistral_chat_message` and `_convert_mistral_chat_message_to_message` returns the original message identically?